### ISSUE TITLE
TSServer: Rename errors on non-existent property

### ISSUE THREAD

**Issue by mjbvz** [Microsoft/TypeScript#27373]
From https://github.com/Microsoft/vscode/issues/59426

**TypeScript Version:**  3.2.0-dev.20180926

**Search Terms:**
- rename
- tsserver 

**Code**
In a TS file:

```ts
class Foo  {
    constructor() {
        this.bar = 1;
    }
}
```

Trigger a rename on `bar`

**bug**
The rename request fails with:

```
[Trace  - 11:46:15 AM] Sending request: rename (247). Response expected: yes. Current queue length: 0
Arguments: {
    "file": "/Users/matb/projects/san/a.ts",
    "line": 3,
    "offset": 15,
    "findInStrings": false,
    "findInComments": false
}
[Trace  - 11:46:15 AM] Response received: rename (247). Request took 2 ms. Success: false . Message: Error processing request. Cannot read property 'start' of undefined
TypeError: Cannot read property 'start' of undefined
    at IOSession.Session.toLocationTextSpan (/Users/matb/projects/san/node_modules/typescript/lib/tsserver.js:122830:65)
    at IOSession.Session.mapRenameInfo (/Users/matb/projects/san/node_modules/typescript/lib/tsserver.js:122428:242)
    at IOSession.Session.getRenameLocations (/Users/matb/projects/san/node_modules/typescript/lib/tsserver.js:122423:39)
    at Session.handlers.ts.createMapFromTemplate._a.(anonymous function) (/Users/matb/projects/san/node_modules/typescript/lib/tsserver.js:121655:61)
    at /Users/matb/projects/san/node_modules/typescript/lib/tsserver.js:123133:88
    at IOSession.Session.executeWithRequestId (/Users/matb/projects/san/node_modules/typescript/lib/tsserver.js:123124:28)
    at IOSession.Session.executeCommand (/Users/matb/projects/san/node_modules/typescript/lib/tsserver.js:123133:33)
    at IOSession.Session.onMessage (/Users/matb/projects/san/node_modules/typescript/lib/tsserver.js:123153:35)
    at Interface.<anonymous> (/Users/matb/projects/san/node_modules/typescript/lib/tsserver.js:124417:27)
    at emitOne (events.js:116:13)
    at Interface.emit (events.js:211:7)
    at Interface._onLine (readline.js:282:10)
    at Interface._normalWrite (readline.js:424:12)
    at Socket.ondata (readline.js:141:10)
    at emitOne (events.js:116:13)
    at Socket.emit (events.js:211:7)
    at addChunk (_stream_readable.js:263:12)
    at readableAddChunk (_stream_readable.js:250:11)
    at Socket.Readable.push (_stream_readable.js:208:10)
    at Pipe.onread (net.js:594:20)
[Trace  - 11:46:15 AM] Sending request: getCodeFixes (248). Response expected: yes. Current queue length: 0
Arguments: {
    "file": "/Users/matb/projects/san/a.ts",
    "startLine": 3,
    "startOffset": 14,
    "endLine": 3,
    "endOffset": 17,
    "errorCodes": [
        2339
    ]
}
[Error  - 11:46:15 AM] 'rename' request failed with error.
Error processing request. Cannot read property 'start' of undefined
TypeError: Cannot read property 'start' of undefined
    at IOSession.Session.toLocationTextSpan (/Users/matb/projects/san/node_modules/typescript/lib/tsserver.js:122830:65)
    at IOSession.Session.mapRenameInfo (/Users/matb/projects/san/node_modules/typescript/lib/tsserver.js:122428:242)
    at IOSession.Session.getRenameLocations (/Users/matb/projects/san/node_modules/typescript/lib/tsserver.js:122423:39)
    at Session.handlers.ts.createMapFromTemplate._a.(anonymous function) (/Users/matb/projects/san/node_modules/typescript/lib/tsserver.js:121655:61)
    at /Users/matb/projects/san/node_modules/typescript/lib/tsserver.js:123133:88
    at IOSession.Session.executeWithRequestId (/Users/matb/projects/san/node_modules/typescript/lib/tsserver.js:123124:28)
    at IOSession.Session.executeCommand (/Users/matb/projects/san/node_modules/typescript/lib/tsserver.js:123133:33)
    at IOSession.Session.onMessage (/Users/matb/projects/san/node_modules/typescript/lib/tsserver.js:123153:35)
    at Interface.<anonymous> (/Users/matb/projects/san/node_modules/typescript/lib/tsserver.js:124417:27)
    at emitOne (events.js:116:13)
    at Interface.emit (events.js:211:7)
    at Interface._onLine (readline.js:282:10)
    at Interface._normalWrite (readline.js:424:12)
    at Socket.ondata (readline.js:141:10)
    at emitOne (events.js:116:13)
    at Socket.emit (events.js:211:7)
    at addChunk (_stream_readable.js:263:12)
    at readableAddChunk (_stream_readable.js:250:11)
    at Socket.Readable.push (_stream_readable.js:208:10)
    at Pipe.onread (net.js:594:20)
```

**Comment by mjbvz** [comment]
@RyanCavanaugh We hit this one pretty often in my testing. I think a fix should go into 3.1.5

**Comment by DanielRosenwasser** [comment]
3.1 fix is in cd1803fdf8fe7061fbe97a154014d0238ccf5551
